### ISSUE TITLE
Regression when used with nock

### ISSUE THREAD

**Issue by lpinca** [sindresorhus/got#1189]
#### Describe the bug

 - Node.js version: v14.0.0
 - OS & version: macOS 10.15.4

`got@11` introduced a regression when used in a testing environment with [nock](https://github.com/nock/nock).

#### Code to reproduce

```js
const assert = require('assert');
const got = require('got');
const nock = require('nock');

nock('http://example.com').get('/test').delayBody(200).reply(200, 'foo');

got('http://example.com/test', { timeout: 100, retry: 0 })
  .then(function () {
    throw new Error('Test invalidation');
  })
  .catch(function (err) {
    assert.strictEqual(err.constructor, got.TimeoutError);
  });
```

Run with `node --unhandled-rejections=strict index.js`

#### Actual behavior

```
$ node --unhandled-rejections=strict index.js 
events.js:292
      throw er; // Unhandled 'error' event
      ^

Error: socket hang up
    at InterceptedRequestRouter.handleAbort (/Users/luigi/test-case/node_modules/nock/lib/intercepted_request_router.js:200:28)
    at OverriddenClientRequest.req.abort (/Users/luigi/test-case/node_modules/nock/lib/intercepted_request_router.js:79:35)
    at PromisableRequest._destroy (/Users/luigi/test-case/node_modules/got/dist/source/core/index.js:1013:32)
    at PromisableRequest.destroy (internal/streams/destroy.js:36:8)
    at PromisableRequest.<anonymous> (/Users/luigi/test-case/node_modules/got/dist/source/as-promise/index.js:194:25)
    at Object.onceWrapper (events.js:422:26)
    at PromisableRequest.emit (events.js:327:22)
    at PromisableRequest._beforeError (/Users/luigi/test-case/node_modules/got/dist/source/as-promise/core.js:115:14)
    at OverriddenClientRequest.<anonymous> (/Users/luigi/test-case/node_modules/got/dist/source/core/index.js:761:18)
    at Object.onceWrapper (events.js:422:26)
Emitted 'error' event on OverriddenClientRequest instance at:
    at /Users/luigi/test-case/node_modules/nock/lib/intercepted_request_router.js:108:11
    at processTicksAndRejections (internal/process/task_queues.js:79:11) {
  code: 'ECONNRESET'
}
```

#### Expected behavior

The process exits cleanly with no errors. This is what happens with `got@10`.

#### Checklist

- [x] I have read the documentation.
- [x] I have tried my code with the latest version of Node.js and Got.

**Comment by szmarczak** [comment]
What is the result on Node.js 13?

**Comment by lpinca** [comment]
The same

```
$ docker run -it --rm -v "$PWD":/tmp/test -w /tmp/test node:13-alpine node index.js 
events.js:292
      throw er; // Unhandled 'error' event
      ^

Error: socket hang up
    at InterceptedRequestRouter.handleAbort (/tmp/test/node_modules/nock/lib/intercepted_request_router.js:200:28)
    at OverriddenClientRequest.req.abort (/tmp/test/node_modules/nock/lib/intercepted_request_router.js:79:35)
    at PromisableRequest._destroy (/tmp/test/node_modules/got/dist/source/core/index.js:1013:32)
    at PromisableRequest.destroy (internal/streams/destroy.js:55:8)
    at PromisableRequest.<anonymous> (/tmp/test/node_modules/got/dist/source/as-promise/index.js:194:25)
    at Object.onceWrapper (events.js:422:26)
    at PromisableRequest.emit (events.js:327:22)
    at PromisableRequest._beforeError (/tmp/test/node_modules/got/dist/source/as-promise/core.js:115:14)
    at OverriddenClientRequest.<anonymous> (/tmp/test/node_modules/got/dist/source/core/index.js:761:18)
    at Object.onceWrapper (events.js:422:26)
Emitted 'error' event on OverriddenClientRequest instance at:
    at /tmp/test/node_modules/nock/lib/intercepted_request_router.js:108:11
    at processTicksAndRejections (internal/process/task_queues.js:79:11) {
  code: 'ECONNRESET'
}
```

**Comment by szmarczak** [comment]
This is actually a bug in Nock. Calling `.abort()` after `.destroy()` should not throw an error. But I'll be happy to implement a fix on the Got side for the time being.

**Comment by szmarczak** [comment]
[Unfortunately I cannot fix it in Got because `request.destroyed` is `undefined`.](https://github.com/nock/nock/issues/1981#issuecomment-619545297)

**Comment by szmarczak** [comment]
Fixed in `nock@13.0.0-beta.2`.
